src/features/swap/useFormValidator.ts

```diff
@@ -1,34 +1,102 @@
-      }
-      return {}
+    (values?: SwapFormValues): Promise<FormikErrors<SwapFormValues>> => {
+      return (async () => {
+        if (!values || !values.amount) return { amount: 'Amount Required' }
+        const parsedAmount = parseAmount(values.amount)
+        if (!parsedAmount) return { amount: 'Amount is Invalid' }
+        if (parsedAmount.lt(0)) return { amount: 'Amount cannot be negative' }
+        if (parsedAmount.lt(MIN_ROUNDED_VALUE)) return { amount: 'Amount too small' }
+        const tokenId = values.fromTokenId
+        const tokenBalance = balances[tokenId]
+        const weiAmount = toWei(parsedAmount, Tokens[values.fromTokenId].decimals)
+        if (weiAmount.gt(tokenBalance) && !areAmountsNearlyEqual(weiAmount, tokenBalance)) {
+          return { amount: 'Amount exceeds balance' }
+        }
+        const { exceeds, errorMsg } = await CheckTradingLimits(values, chainId)
+        if (exceeds) return { amount: errorMsg }
+        return {}
+      })().catch((error) => {
+        logger.error(error)
+        return {}
+      })
     },
-    [balances, sizeLimits]
+    [balances, chainId]
   )
 }
+
+async function CheckTradingLimits(
+  values: SwapFormValues,
+  chainId: number
+): Promise<{ exceeds: boolean; errorMsg: string }> {
+  const token0 = getTokenAddress(values.fromTokenId, chainId)
+  const token1 = getTokenAddress(values.toTokenId, chainId)
+  const mento = await getMentoSdk(chainId)
+  const exchangeId = (await mento.getExchangeForTokens(token0, token1)).id
+  const tradingLimits = await mento.getTradingLimits(exchangeId)
+
+  let timestampIn = 0
+  let timestampOut = 0
+  let minMaxIn = Infinity
+  let minMaxOut = Infinity
+
+  for (const limit of tradingLimits) {
+    if (limit.maxIn < minMaxIn) {
+      minMaxIn = limit.maxIn
+      timestampIn = limit.until
+    }
+    if (limit.maxOut < minMaxOut) {
+      minMaxOut = limit.maxOut
+      timestampOut = limit.until
+    }
+  }
+  const isSwapIn = values.direction === 'in'
+  const tokenToCheck = getTokenByAddress(tradingLimits[0].asset).symbol
+
+  let amountToCheck: number
+  let exceeds = false
+  let limit = 0
+  let timestamp = 0
+
+  if (tokenToCheck === values.fromTokenId) {
+    amountToCheck = isSwapIn ? +values.amount : +values.quote
+    if (amountToCheck > minMaxIn) {
+      exceeds = true
+      limit = minMaxIn
+      timestamp = timestampIn
+    }
+  } else {
+    amountToCheck = isSwapIn ? +values.quote : +values.amount
+    if (amountToCheck > minMaxOut) {
+      exceeds = true
+      limit = minMaxOut
+      timestamp = timestampOut
+    }
+  }
+
+  const date = new Date(timestamp * 1000).toLocaleString()
+
+  if (exceeds) {
+    const errorMsg = `The ${tokenToCheck} amount exceeds the current trading limits. The current ${
+      tokenToCheck === values.fromTokenId ? 'sell' : 'buy'
+    }  limit is ${new Intl.NumberFormat('de-DE').format(limit)} ${tokenToCheck} until ${date}`
```

Comment:
I don't think we should default to de-DE format here. The rest of the app uses US-EN format, which is typically the default until an app is fully i18nized

Reply:
I used the de-De version because the values in the form also follow this format in the sense that we use the , to separate the decimal part. But  I just saw that for the price `2.0970 CELO ~ 1 cUSD` we use the . 
I think we should make this consistent across the app

Reply:
Agreed! I'm open to either format as along as we use it consistently :)

Reply:
I created a ticket for that https://github.com/mento-protocol/mento-web/issues/85